### DISCUSSION TITLE
How can multiple preview environments be created on the same branch?

### DISCUSSION THREAD

**Discussion by yairaba** [argoproj/argo-cd#18638] [General]
Hi All,
How can multiple preview environments be created on the same branch?
The following will create one preview environment when I open a pull request from branch1 to master (env1 of branch1)
What if I want to create an environment 2 of branch1
Thanks

apiVersion: argoproj.io/v1alpha1
kind: ApplicationSet
metadata:
name: fb-matrix
spec:
goTemplate: true
generators:
- matrix:
generators:
- pullRequest:
github:
owner: name
repo: argo-plugin
labels:
- preview-matrix
tokenRef:
secretName: github-secret
key: token
requeueAfterSeconds: 30
- plugin:
configMapRef:
name: cm-plugin
input:
parameters:
repositories:
- repo: "repoBack"
branches: ['{{.branch}}']
- repo: "repoFront"
branches: ['branch2']
requeueAfterSeconds: 10
template:
metadata:
name: "fb-matrix-{{.branch}}-1"
spec:
source:
repoURL: "https://github.com/name/argo-plugin.git"
targetRevision: "HEAD"
path: charts/app-client
helm:
releaseName: feature-test-matrix-{{.branch}}
valueFiles:
- values.yaml
values: |
front:
image: hub.docker.com/repositories/name/front:{{ .serviceData.digitFront }}
back:
image: hub.docker.com/repositories/name/back:{{ .serviceData.digitBack }}
project: default
syncPolicy:
automated:
prune: true
selfHeal: true
syncOptions:
- CreateNamespace=true
destination:
name: in-cluster
namespace: 'pr-{{.branch}}'

**Comment by kostis-codefresh** [comment]
Please format the example as YAML

Also the accepted practice is one branch per PR. Can you explain a bit what you are trying to do? Does GitHub actually allow you to create two separate PRs for the same branch?
